### PR TITLE
feat: deprecate py38, support py11 py12

### PR DESCRIPTION
**Pull request by ovesh**
Support py311 change to PyType_Ready()
see https://docs.python.org/3/whatsnew/3.11.html
fixes 'type genome_kit._cxx.GenomeTrackBuilder has the Py_TPFLAGS_HAVE_GC flag but has no traverse function'

Fix sigsegv on exit on py>=312, see https://github.com/python/cpython/issues/126508

also:
- remove GenomeTrack unused owner ref
- fail unit tests if this python version wasn't built due to feedstock's meta.yaml
- remove unused numpy pin from conda_build_config.yaml
- numpy <2.0dev0 to match feedstock's meta.yaml (fix build errors)

### CODE REVIEW

**Comment by ovesh** [comment]
I went over py311 and py12 release notes and didn't find anything that requires us to update the C code.

**Comment by s22chan** [comment]
```python
>>> python -c 'import genome_kit as gk'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/data/steve/mambaforge/envs/py312/lib/python3.12/site-packages/genome_kit/__init__.py", line 6, in <module>
    from . import gk_data
  File "/data/steve/mambaforge/envs/py312/lib/python3.12/site-packages/genome_kit/gk_data.py", line 29, in <module>
    from . import _cxx
  File "<frozen importlib._bootstrap>", line 649, in parent
SystemError: type genome_kit._cxx.GenomeTrackBuilder has the Py_TPFLAGS_HAVE_GC flag but has no traverse function
```

**Comment by ovesh** [comment]
Thanks @s22chan. Found this in the [Python3.11 release notes](https://docs.python.org/3/whatsnew/3.11.html):
> The [PyType_Ready()](https://docs.python.org/3/c-api/type.html#c.PyType_Ready) function now raises an error if a type is defined with the [Py_TPFLAGS_HAVE_GC](https://docs.python.org/3/c-api/typeobj.html#c.Py_TPFLAGS_HAVE_GC) flag set but has no traverse function ([PyTypeObject.tp_traverse](https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_traverse)). (Contributed by Victor Stinner in [bpo-44263](https://bugs.python.org/issue?@action=redirect&bpo=44263).)

**Comment by ovesh** [comment]
Found out that:
* meta.yaml (that the github action downloads from conda-forge) needs to be updated first (`skip: true  # [py>310 or py<38]`)
* `conda mambabuild --croot /tmp/conda-bld -t ~/conda-bld/*/*py311*.tar.bz2 --extra-deps python=3.11` will exit 0 and print "All tests passed" even if it can't find a matching tarball for that version of python

**Comment by ovesh** [comment]
I'll make a separate PR for deprecating py38 to unblock #98 (failing py38 tests).

**Comment by s22chan** [comment]
> I'll make a separate PR for deprecating py38 to unblock #98 (failing py38 tests).

#98 is a different issue

**Review by s22chan — APPROVED** [review]
dare to add py313?

File: setup.py
Comment (s22chan) [inline setup.py:38]: not sure when 3.8/3.7 was removed, but technically the setuptools still supports 3.7/3.8?

**Comment by ovesh** [comment]
> dare to add py313?

Got numpy errors last time I tried, I'm going to hold off until after this is merged.

File: setup.py
Comment (ovesh) [reply setup.py:38]: Good point, but I'm not sure that we want to declare support for 3.8 if we're not running unit tests for it.

**Pull request merged**
